server/test/trigger/trigger.go

```diff
@@ -0,0 +1,61 @@
+package trigger
+
+import "encoding/json"
+
+type (
+	TriggerType string
+
+	Trigger struct {
+		Type    TriggerType     `json:"type"`
+		HTTP    *HTTPRequest    `json:"http,omitempty"`
+		GRPC    *GRPCRequest    `json:"grpc,omitempty"`
+		TraceID *TraceIDRequest `json:"traceid,omitempty"`
+	}
+
+	TriggerResult struct {
+		Type    TriggerType      `json:"type"`
+		HTTP    *HTTPResponse    `json:"http,omitempty"`
+		GRPC    *GRPCResponse    `json:"grpc,omitempty"`
+		TraceID *TraceIDResponse `json:"traceid,omitempty"`
+	}
+)
+
+// Compatibility with older versions
+type triggerFormat struct {
+	Type    TriggerType     `json:"type"`
+	HTTP    *HTTPRequest    `json:"http,omitempty"`
+	GRPC    *GRPCRequest    `json:"grpc,omitempty"`
+	TraceID *TraceIDRequest `json:"traceid,omitempty"`
+}
+
+type oldTriggerFormat struct {
```

Comment:
again, I think having "versioned" formats makes it easier to understand the chronology: https://github.com/kubeshop/tracetest/blob/main/server/model/trigger_json.go